plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -79,21 +82,22 @@ public void loadConnectors(Settings settings) {
   @Override
   public StorageEngine getStorageEngine(String catalog) {
     if (catalog == null || !storageEngineMap.containsKey(catalog)) {
-      return storageEngineMap.get(OPEN_SEARCH);
+      return defaultOpenSearchStorageEngine;
     }
     return storageEngineMap.get(catalog);
   }
 
   @Override
   public Set<String> getCatalogs() {
-    Set<String> catalogs = storageEngineMap.keySet();
-    catalogs.remove(OPEN_SEARCH);
-    return catalogs;
+    return Collections.unmodifiableSet(storageEngineMap.keySet());
```

Comment:
the original issue was `storageEngineMap.keySet().remove(OPENSEARCH)` modified `storageEngineMap`? also checkstyle failed

Reply:
Yes, don't want to maintain separate entry for defaultOSStorageEngine in storageEngineMap. If there is a reload and new loadConnectors call, we create a new map with new storage engine objects, there is a chance it could lead to defaultOSStorageEngine not available.

Fixing the checkstyle issue.